src/codegen/statements.rs

```diff
@@ -530,21 +518,10 @@ pub(crate) fn statement(
 
             let mut next_reachable = true;
 
-            for stmt in next {
-                statement(
-                    stmt,
-                    func,
-                    cfg,
-                    contract_no,
-                    ns,
-                    vartab,
-                    loops,
-                    placeholder,
-                    return_override,
-                    opt,
-                );
+            if let Some(next) = next {
+                expression(next, cfg, contract_no, Some(func), ns, vartab, opt);
 
-                next_reachable = stmt.reachable();
+                next_reachable = next.ty() != Type::Unreachable;
```

Comment:
Same here, the below if statement could go directly in here? And the same on line 511

Reply:
Again we want the branch also if there is no next